# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/reftests/text-overflow/selection.html
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 layout/reftests/text-overflow/selection.html
--- a/layout/reftests/text-overflow/selection.html	Thu Sep 22 06:24:38 2022 +0000
+++ b/layout/reftests/text-overflow/selection.html	Thu Sep 22 06:27:37 2022 +0000
@@ -82,7 +82,11 @@
 }
 }
 function selectText() {
- var divs = document.getElementsByTagName('div');
+  // For putting selection ranges to multiple editing hosts, we need to put
+  // a range outside all editable elements first, then, selection's ancestor
+  // limiter won't be set to an editing host.
+  getSelection().collapse(document.body, document.body.childNodes.length);
+  var divs = document.getElementsByTagName('div');
   for (i = 0; i < divs.length; ++i) {
     addRange(divs[i]);
   }